wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,196 @@
+        List<WorkspaceImpl> workspaces;
+        try {
+            workspaces = workspaceManager.getWorkspaces(user.getId());
+        } catch (ServerException e) {
+            printRowToReport("Failed to receive list of related workspaces for user " + user.getEmail() + "!\n", file);
+            LOG.error(e.getMessage(), e);
+            return;
+        }
+
+        int workspacesNumber = workspaces.size();
+        long ownWorkspacesNumber = workspaces.stream().filter(workspace -> workspace.getNamespace().equals(user.getName())).count();
+        printRowToReport(user.getEmail() + " is owner of " +
+                         ownWorkspacesNumber + " workspace" + (ownWorkspacesNumber > 1 | ownWorkspacesNumber == 0 ? "s" : "") +
+                         " and has permissions in " + workspacesNumber + " workspace" +
+                         (workspacesNumber > 1 | workspacesNumber == 0 ? "s" : "") + "\n", file);
+        for (WorkspaceImpl workspace : workspaces) {
+            printUserWorkspaceInfo(workspace, user, file);
+        }
+    }
+
+    private void printUserWorkspaceInfo(WorkspaceImpl workspace, UserImpl user, File file) throws ServerException {
+        printRowToReport("   └ " + workspace.getConfig().getName() +
+                         ", is owner: " + workspace.getNamespace().equals(user.getName()) + ", permissions: ", file);
+        try {
+            printRowToReport(getWorkspacePermissions(workspace.getId(), user.getId()).getActions().toString() + "\n", file);
+        } catch (NotFoundException e) {
+            printRowToReport("Failed to retrieve workspace permissions! " + e.getMessage() + "\n", file);
+        } catch (ConflictException e) {
+            LOG.error(e.getMessage(), e);
+            printRowToReport("Failed to retrieve workspace Id!\n", file);
+        }
+    }
+
+    private void printRowToReport(String row, File report) throws ServerException {
+        try {
+            Files.append(row, report, Charset.defaultCharset());
+        } catch (IOException e) {
+            LOG.error(e.getMessage(), e);
+            throw new ServerException("Failed to generate audit report. " + e.getMessage(), e);
+        }
+    }
+
+    private PermissionsImpl getWorkspacePermissions(String workspaceId, String userId)
+            throws ServerException, NotFoundException, ConflictException {
+        List<PermissionsImpl> permissions;
+        permissions = permissionManager.getByInstance("workspace", workspaceId);
+        Optional<PermissionsImpl> optional = permissions.stream()
+                                                        .filter(wsPermissions -> wsPermissions.getUser().equals(userId))
+                                                        .findFirst();
+
+        if (optional.isPresent()) {
+            return optional.get();
+        } else {
+            throw new NotFoundException("Permissions for user " + userId + " in workspace " + workspaceId +
```

Comment:
Actually the report WILL be generated. The exception is caught in `printUserWorkspaceInfo` method, it prints error in the row where permissions are expected to be.